### PR TITLE
New packging for SAMLv2 addon

### PR DESCRIPTION
**Pull request by hbenali**

### CODE REVIEW

File: packaging/src/main/assemblies/exo-saml-addon.xml [hunk 1]
```diff
@@ -53,6 +45,7 @@
         <include>org.picketlink:picketlink-common</include>
         <include>org.picketlink:picketlink-config</include>
         <include>org.picketlink:picketlink-federation</include>
+        <include>org.picketbox:jboss-security-spi</include>
         <include>org.picketlink.distribution:picketlink-wildfly8</include>
```
Comment (boubaker) [inline packaging/src/main/assemblies/exo-saml-addon.xml]: why keeping picketlink-wildfly8 artifact ?

Comment (hbenali) [reply packaging/src/main/assemblies/exo-saml-addon.xml]: I have removed it

File: pom.xml [hunk 2]
```diff
@@ -108,6 +108,11 @@
         <artifactId>picketlink-config</artifactId>
         <version>${version.picketlink.fed}</version>
       </dependency>
+      <dependency>
+        <groupId>org.picketbox</groupId>
+        <artifactId>jboss-security-spi</artifactId>
+        <version>3.0.0.Final</version>
```
[the remark this answers is not in the document]
Comment (hbenali) [reply pom.xml]: In fact, there is no jboss-security-spi with 2.5.5.Final version.

File: packaging/src/main/resources/exo-saml2-config/conf/Catalina/portal.xml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<!--
```
Comment (boubaker) [inline packaging/src/main/resources/exo-saml2-config/conf/Catalina/portal.xml]: This file already exists in https://github.com/exodev/gatein-portal/blob/develop/web/portal/src/main/webapp/META-INF/context.xml#L22 and it has already variables that uses exo.properties

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (boubaker) [reply pom.xml]: I mean, it should goes here:
https://github.com/exo-addons/saml2-addon/pull/3/files#diff-600376dffeb79835ede4a0b285078036R49

[the remark this answers is not in the document]
Comment (hbenali) [reply pom.xml]: Done

File: packaging/src/main/resources/exo-saml2-config/conf/Catalina/portal.xml (hunk 3 above)
Comment (hbenali) [reply packaging/src/main/resources/exo-saml2-config/conf/Catalina/portal.xml]: File removed

File: README.md [hunk 4]
```diff
@@ -12,51 +9,63 @@ SAML2 SP integration works only for eXo Platform Tomcat
...
-gatein.sso.idp.keystorepass=store123
 gatein.sso.filter.logout.enabled=true
 gatein.sso.filter.logout.class=org.gatein.sso.saml.plugin.filter.SAML2LogoutFilter
 gatein.sso.filter.logout.url=${gatein.sso.sp.url}?GLO=true 
+
+# Custom properties
+
+gatein.sso.sp.host=SP_HOSTNAME
+gatein.sso.sp.url=${gatein.sso.sp.host}/portal/dologin
+gatein.sso.idp.host=IDP_HOSTNAME
```
Comment (boubaker) [inline README.md]: is this useless ?

File: README.md [hunk 5]
```diff
@@ -12,51 +9,63 @@ SAML2 SP integration works only for eXo Platform Tomcat
...
+gatein.sso.idp.alias=IDP_SIGNING_ALIAS
+gatein.sso.idp.signingkeypass=IDP_SIGNING_KEY_PASS
+gatein.sso.idp.keystorepass=IDP_KEYSTORE_PASS
 # WARNING: This bundled keystore is only for testing purposes. You should generate and use your own keystore!
 gatein.sso.picketlink.keystore=${exo.conf.dir}/saml2/jbid_test_keystore.jks
-
+```
+Note: The following properties values must be configured
+```
+SP_HOSTNAME: eXo SP Hostname, Example: sp.com
```
Comment (boubaker) [inline README.md]: Example: **https://**sp.com

File: README.md [hunk 6]
```diff
@@ -12,51 +9,63 @@ SAML2 SP integration works only for eXo Platform Tomcat
...
+IDP_HOSTNAME: Saml IDP Hostname, Example, idp.com
```
Comment (boubaker) [inline README.md]: useless IMO

File: README.md [hunk 7]
```diff
@@ -12,51 +9,63 @@ SAML2 SP integration works only for eXo Platform Tomcat
...
+IDP_SAML_ENDPOINT: Saml IDP Endpoint: Example, http://idp.com/saml
+IDP_SIGNING_ALIAS: SSL IDP Alias, Example: idpalias
```
Comment (boubaker) [inline README.md]: please add more details about this alias, like (certificate alias in selected Keystore file)

File: README.md [hunk 8]
```diff
@@ -12,51 +9,63 @@ SAML2 SP integration works only for eXo Platform Tomcat
...
+IDP_SIGNING_KEY_PASS: SSL Signging Password Key, Example: test123
+IDP_KEYSTORE_PASS: SSL Keystore Password, Example: store123
```
Comment (boubaker) [inline README.md]: "Certificates Keystore password" instead of "SSL Keystore Password"

File: README.md (hunk 4 above)
Comment (hbenali) [reply README.md]: No, it is required for picketlink-sp.xml file (gatein.sso.idp.host)

File: README.md (hunk 5 above)
Comment (hbenali) [reply README.md]: Done

File: README.md (hunk 7 above)
Comment (hbenali) [reply README.md]: Done

File: README.md (hunk 8 above)
Comment (hbenali) [reply README.md]: Done

File: README.md (hunk 6 above)
Comment (hbenali) [reply README.md]: Well, it is just for indication IMO

Comment (boubaker) [reply README.md]: it's better to optimize the number of parameters to introduce

Comment (hbenali) [reply README.md]: OK

**Pull request merged**